### PR TITLE
Create UI Search

### PR DESCRIPTION
**Pull request by NguyenNgocTuyenUED**
Create UI Search + Validate Video Data
https://edu-redmine.sun-asterisk.vn/issues/8869
https://edu-redmine.sun-asterisk.vn/issues/8871
image

### CODE REVIEW

File: app/src/main/java/com/sun_asterisk/youtubebackground/screen/search/SearchFragment.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    private SearchContract.Presenter mPresenter;
+    private VideoRepository mVideoRepository;
+    private ListView mListViewSearch;
+
+    public static SearchFragment newInstance() {
+        return new SearchFragment();
+    }
+
+    @Override
+    public void onGetVideoSuccess(List<Video> video) {
```
Comment (thaohq-0814) [inline app/src/main/java/com/sun_asterisk/youtubebackground/screen/search/SearchFragment.java]:
```suggestion
    public void onGetVideoSuccess(List<Video> videos) {
```

File: app/src/main/java/com/sun_asterisk/youtubebackground/screen/search/SearchFragment.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            resetSearch();
+            return false;
+        }
+        List<String> filteredValues = new ArrayList<String>(mTitles);
+        for (String value : mTitles) {
+            if (!value.toLowerCase().contains(newText.toLowerCase())) {
+                filteredValues.remove(value);
+            }
+        }
+        mAdapter = new ArrayAdapter<>(getActivity(), android.R.layout.simple_list_item_1,
```
Comment (thaohq-0814) [inline app/src/main/java/com/sun_asterisk/youtubebackground/screen/search/SearchFragment.java]: use NotifyDataSetChange() to update data, don't create new Object

File: app/src/main/java/com/sun_asterisk/youtubebackground/screen/search/SearchFragment.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                filteredValues);
+        mListViewSearch.setAdapter(mAdapter);
+        return false;
+    }
+
+    private void resetSearch() {
+        mAdapter = new ArrayAdapter<>(getActivity(), android.R.layout.simple_list_item_1, mTitles);
```
Comment (thaohq-0814) [inline app/src/main/java/com/sun_asterisk/youtubebackground/screen/search/SearchFragment.java]: as above

File: app/src/main/java/com/sun_asterisk/youtubebackground/screen/search/SearchFragment.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        searchView.setOnQueryTextListener(this);
+        super.onCreateOptionsMenu(menu, inflater);
+    }
+
+    private void listAllTitle() {
+        mTitles = new ArrayList<>();
+        for (int i = 0; i < 20; i++) {
+            mTitles.add(mVideos.get(i).getTitle());
+        }
+        mAdapter = new ArrayAdapter<>(getActivity(), android.R.layout.simple_list_item_1, mTitles);
```
Comment (thaohq-0814) [inline app/src/main/java/com/sun_asterisk/youtubebackground/screen/search/SearchFragment.java]: initialize Adapter in  onCreate().

File: app/src/main/java/com/sun_asterisk/youtubebackground/screen/search/SearchFragment.java [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    private VideoRepository mVideoRepository;
+    private ListView mListViewSearch;
+
+    public static SearchFragment newInstance() {
+        return new SearchFragment();
+    }
+
+    @Override
+    public void onGetVideoSuccess(List<Video> video) {
+        mVideos = new ArrayList<>();
```
Comment (thaohq-0814) [inline app/src/main/java/com/sun_asterisk/youtubebackground/screen/search/SearchFragment.java]: initialize it in onCreate().

File: app/src/main/java/com/sun_asterisk/youtubebackground/screen/search/SearchFragment.java [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+            Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_search, container, false);
+        mVideoRepository = VideoRepository.getInstance(VideoRemoteDataSource.getInstance());
+        mPresenter = new SearchPresenter(this, mVideoRepository);
+        mPresenter.getVideos();
+        mListViewSearch = view.findViewById(R.id.listViewSearch);
+        mListViewSearch.setAdapter(mAdapter);
```
Comment (thaohq-0814) [inline app/src/main/java/com/sun_asterisk/youtubebackground/screen/search/SearchFragment.java]: init mAdapter before set it into ListView

File: app/src/main/res/layout/fragment_search.xml [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@android:color/background_light"
+    >
+
+    <ListView
+        android:id="@+id/listViewSearch"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        ></ListView>
```
Comment (thaohq-0814) [inline app/src/main/res/layout/fragment_search.xml]:
```suggestion
        />
```

**Review by thaohq-0814 — APPROVED** [review]

**Comment by NguyenNgocTuyenUED** [comment]
ready

File: app/src/main/res/layout/fragment_search.xml [hunk 8]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.constraint.ConstraintLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@android:color/background_light"
+    >
+
+    <ListView
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/fragment_search.xml]: sao lại dùng listview

File: app/src/main/java/com/sun_asterisk/youtubebackground/utils/Navigator.java [hunk 9]
```diff
@@ -37,6 +38,19 @@ public void addFragment(FragmentActivity fragmentActivity, Fragment fragment, in
                     fragmentActivity.getSupportFragmentManager().popBackStack(PLAY, 0);
                 }
                 break;
+            case R.layout.fragment_search:
+                FragmentTransaction fragmentTransactionSearch =
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun_asterisk/youtubebackground/utils/Navigator.java]: tách method ra e

File: app/src/main/java/com/sun_asterisk/youtubebackground/screen/search/SearchFragment.java [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+
+public class SearchFragment extends Fragment
+        implements SearchView.OnQueryTextListener, MenuItem.OnActionExpandListener,
+        SearchContract.View {
+    private List<Video> mVideos;
+    private List<String> mTitles;
+    private ArrayAdapter<String> mAdapter;
+    private SearchContract.Presenter mPresenter;
+    private VideoRepository mVideoRepository;
+    private ListView mListViewSearch;
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun_asterisk/youtubebackground/screen/search/SearchFragment.java]: sử dụng RecyclerView e

**Comment by NguyenNgocTuyenUED** [comment]
ready

File: app/src/main/java/com/sun_asterisk/youtubebackground/screen/search/SearchFragment.java [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setHasOptionsMenu(true);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+            Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_search, container, false);
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun_asterisk/youtubebackground/screen/search/SearchFragment.java:60]: tách method ra e

File: app/src/main/res/layout/search_item.xml [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+    android:layout_height="wrap_content">
+
+    <TextView
+        android:id="@+id/textTitleVideo"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:textSize="@dimen/sp_16"
+        tools:ignore="MissingConstraints"
+        />
+</android.support.constraint.ConstraintLayout>
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/search_item.xml]: enter end line

File: app/src/main/java/com/sun_asterisk/youtubebackground/screen/search/SearchFragment.java [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+        return false;
+    }
+
+    @Override
+    public boolean onQueryTextChange(String newText) {
+        if (newText == null || newText.trim().isEmpty()) {
+            resetSearch();
+            return false;
+        }
+        List<String> filteredValues = new ArrayList<String>(mTitles);
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun_asterisk/youtubebackground/screen/search/SearchFragment.java]: logic chỗ này không ổn e

**Comment by NguyenNgocTuyenUED** [comment]
ready

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**